charts/datadog/templates/_containers-common-env.yaml

```
@@ -87,6 +88,14 @@
 {{- end }}
 {{- end }}
 {{- end }}
+
+{{- with .Values.datadog.otlp.logs }}
+
+- name: DD_OTLP_CONFIG_LOGS_ENABLED
+  value: {{ .enabled | quote }}
+
+{{- end }}
```

small nit for code clarity: remove the 2 new line between the `{{- with ... }} ... {{- end }}` condition

```suggestion
{{- with .Values.datadog.otlp.logs }}
- name: DD_OTLP_CONFIG_LOGS_ENABLED
  value: {{ .enabled | quote }}
{{- end }}
```